containerd-shim-spin/src/engine.rs

```diff
@@ -156,47 +159,53 @@ impl SpinEngine {
         app: LockedApp,
         app_source: Source,
     ) -> Result<()> {
+        let mut loader = spin_trigger::loader::ComponentLoader::default();
+        match app_source {
+            Source::Oci => unsafe {
+                // Configure the loader to support loading AOT compiled components..
+                // Since all components were compiled by the shim (during `precompile`),
+                // this operation can be considered safe.
+                loader.enable_loading_aot_compiled_components();
+            },
+            // Currently, it is only possible to precompile applications distributed using
+            // `spin registry push`
+            Source::File(_) => {}
+        };
         let mut futures_list = Vec::with_capacity(trigger_types.len());
         for trigger_type in trigger_types.iter() {
+            let app = spin_app::App::new("TODO", app.clone());
```

Comment:
This seems to be something we might be able to change in upstream as that app id is rarely actually used. Currently, in the Spin code base it's only being used in the Redis trigger for telemetry purposes. It's used by some other runtimes simply as a unique identifier.

Reply:
By globally do you mean unique to the cluster too? Container name may be a good option

Reply:
The uniqueness isn't necessary (as far as I can tell) for everything to work properly - it is just needed for proper diagnostics as this string might be exposed to the user in telemetry. Therefore, I think we want something that makes sense to users if they see it. 

I'm not sure what the best thing to use is - if you say container name works well that's fine by me. How do I access that information?

Reply:
We should be able to use the `HOSTNAME` env var to get the container name. This is not a unique container id; rather just the container name (ie `busybox`). Furthermore, if the container is using `hostNetwork`, the hostname becomes the node hostname, but in production, that is rarely done. For now `HOSTNAME` sounds like the best bet

Reply:
Did a quick check locally, printing the value of `HOSTNAME` from the shim and looks like it is the fully unique container name: `"container name: simple-spinapp-59c4d9668-gz8dr"`

Reply:
@rylev what are your thoughts on using `HOSTNAME`?

Reply:
I don't have a better idea. I suppose this isn't hard to change in the future and since nothing is really relying on the id being a particular format (it's just to help humans), we don't really even need to worry about backwards compatibility.